LoginClient/LoginClientForm.cs

```diff
@@ -0,0 +1,230 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+/// <summary>
+/// Author: Andrew Williamson
+/// Student ID: P113357
+/// 
+/// AT 2 - Question 4 
+/// 
+/// JMC wishes to have a standard login functionality for all their 
+/// terminals around the ship, this should be accomplished via logging 
+/// into a central server to test user and password combinations 
+/// (you must have at least one administrator password setup)
+/// You must create a Server Client program it must use IPC to communicate.
+/// Your program must have a login that uses standard hashing techniques.
+/// 
+/// </summary>
+namespace LoginClient
+{
+    public partial class LoginClientForm : Form
+    {
+        private PipeClient pipeClient;
+
+        public LoginClientForm()
+        {
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// This method sets up the program when it first loads.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void LoginClientForm_Load(object sender, EventArgs e)
+        {
+            CreateNewPipeClient();
+        }
+
+        /// <summary>
+        /// Create a new pipeClient and set all the handlers for the delegates.
+        /// </summary>
+        private void CreateNewPipeClient()
+        {
+            if (pipeClient != null)
+            {
+                pipeClient.MessageRecieved -= pipeClient_MessageRecieved;
+                pipeClient.ServerDisconnected -= pipeClient_ServerDisconnected;
+                pipeClient.ConnectedToServer -= pipeClient_ConnectedToServer;
+            }
```

Comment:
You dont tell the client to disconnect here? So does it disconnect before this function? 
Easier if "All" connection processes start from the same place. CreateNewPipeClient() is a good place for them to all start.